libsolidity/analysis/TypeChecker.cpp

```diff
@@ -193,6 +193,18 @@ TypePointers TypeChecker::typeCheckABIDecodeAndRetrieveReturnType(FunctionCall c
 					typeArgument->location(),
 					"Decoding type " + actualType->toString(false) + " not supported."
 				);
+
+			if (auto referenceType = dynamic_cast<ArrayType const*>(actualType))
```

Comment:
Why only for array type and not for all reference types?

Reply:
Good point. Will change.